### ISSUE TITLE
FileNotFound Exception in gRPC C# 0.13.0 when used for webapplication

### ISSUE THREAD

**Issue by cw-gawdeprasad** [grpc/grpc#5589]
I updated my `gPRC `version for the WebApplication from `0.12.0` to `0.13.0` and it started failing to create a `Channel `object. Following is the exception thrown:

![image](https://cloud.githubusercontent.com/assets/14074555/13518674/7e9e95ba-e1f6-11e5-999a-40b0b6f0b97c.png)

It is trying to locate grpc_csharp_ext.dll at `AppData\\Local\\Temp\\Temporary ASP.NET Files\\root\\c8490d1a\\9f150f28\\assembly\\dl3\\5d08d985\\2c1c8757_6474d101\\nativelibs\\windows_x86\\grpc_csharp_ext.dll`. I even stored the DLL in local bin folder but that didn't help. This problem does not occur if I create a Console app. I am not sure why it is trying to search the dll at the said location? Does anyone know how to make it work for a web application? As per my discussion with Jan Tattermusch, this seems to be a Grpc Bug so filing it as an issue here.

I have created a stackoverflow question for the same at http://stackoverflow.com/questions/35746910/filenotfound-exception-in-grpc-c-sharp-0-13-0-when-used-for-webapplication

**Comment by jtattermusch** [comment]
I think this might be because  with web applications the dlls are actually getting copied to a different location once you start your app. 

The way the grpc_csharp_ext.dll is expected to be found in a directory windows_x86 or windows_x64 relatively to path to the currently executing assembly.

With console app, everything works because the Grpc.Core.dll is not copied anywhere and the grpc_csharp_ext.dll is discovered correctly.

See this code:
https://github.com/grpc/grpc/blob/master/src/csharp/Grpc.Core/Internal/NativeExtension.cs#L101

Basically there's two possible solutions:
1. convince the webserver to also copy windows_x86 and windows_x64 directories when shadowing the Grpc.Core.dll assembly
2. tweak the discovery code to see where the unshadowed version of Grpc.Core.dll is located and load grpc_csharp_ext.dll relative to that path.

https://msdn.microsoft.com/en-us/library/ms404279(v=vs.110).aspx

**Comment by Sildra** [comment]
duplicate of https://github.com/grpc/grpc/issues/5425

**Comment by cw-gawdeprasad** [comment]
Thanks a lot jtattermusch. For time being I have changed the code in NAtiveExtension.cs to read the corresponding file location from appsetting. And it is working fine for me. I just need to give read and execute permission to the folder which holds grpc_csharp_ext.dll. Once this issue is fixed (hopefully in next release), I will get take the new changes.

**Comment by jtattermusch** [comment]
@Sildra , it's not really a duplicate of #5425 but it's related.

**Comment by jtattermusch** [comment]
gRPC C# 0.13.1 have been released, can you please confirm that this is fixed?

**Comment by Sildra** [comment]
I confirm that the issue has been resolved. Thanks.

**Comment by cw-gawdeprasad** [comment]
I am still getting the exception. Please see the following screenshot

![image](https://cloud.githubusercontent.com/assets/14074555/14427085/6ca0d19c-0010-11e6-97a5-f279c385e955.png)


I simply created a sample website, installed the latest GRPC C# from Nuget manager and tried creating a Channel

**Comment by cw-gawdeprasad** [comment]
It worked when I gave the execute right to Network Service (I am running the website with the app pool identity of Network Service). Thanks a lot.
